File: fixtures/matlab_output/post_basic_auth_url_encoded_data.m
%% Web Access using Data Import and Export API
uri = 'http://localhost/api/oauth/token/';
body = 'grant_type=client_credentials';
options = weboptions(...
    'Username', 'foo',...
    'Password', 'bar'...
);
response = webwrite(uri, body, options);

%% HTTP Interface
import matlab.net.*
import matlab.net.http.*
import matlab.net.http.io.*

uri = URI('http://localhost/api/oauth/token/');
cred = Credentials('Username', 'foo', 'Password', 'bar');
options = HTTPOptions('Credentials', cred);
body = FormProvider('grant_type', 'client_credentials');
response = RequestMessage('post', [], body).send(uri.EncodedURI, options);

File: test/fixtures/node-axios/post_basic_auth_url_encoded_data.js
import axios from 'axios';

const response = await axios.post(
  'http://localhost:28139/api/oauth/token/',
  new URLSearchParams({
    'grant_type': 'client_credentials'
  }),
  {
    auth: {
      username: 'foo',
      password: 'bar'
    }
  }
);
